SignUp/SignUp/View/CustomViews/CustomTextField.swift

```diff
@@ -0,0 +1,56 @@
+//
+//  CustomTextField.swift
+//  SignUp
+//
+//  Created by 백상휘 on 2022/03/28.
+//
+
+import UIKit
+
+final class CustomTextField: UITextField, UITextFieldDelegate {
+    
+    // MARK: - Local Properties
+    
+    private lazy var commentLabel = CustomCommentLabel(frame: self.frame)
+    var validator: ValidationRegularText?
+    
+    // MARK: - Initialzers
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        self.delegate = self
```

Comment:
델리게이트는 이런 TextField를 만들더라도 뷰의 입출력과 동작을 분리하기 위한 겁니다
그런데 이렇게 내부에서 구현해버리면, TextField와 델리게이트를 구분하는 의미가 없습니다 ㅜㅜ
일부러 유연한 구조 만들수 있는 코드를 억지로 다시 붙여놓으셨네요. 이런 방향은 매우 어색한 상황입니다

Reply:
@godrm 질문이 있습니다!

루카스에 나와있는 대로 델리게이트를 따로 구현하는 클래스로 만들었으나
```Instance will be immediately deallocated because property 'delegate' is 'weak'```
와 같은 Warning을 받아서 결국 제외하게 된 일이 있습니다.

여러 시도를 해보던 중 신기한 일이 있었는데,

```swift
// 이전에 구현한 방식입니다. 
idTextField.delegate = CustomTextFieldDelegate()
print(idTextField.delegate == nil) // true

// 새로 시도해본 방식입니다.
let delegate = CustomTextFieldDelegate()
idTextField.delegate = delegate
print(idTextField.delegate == nil) // false
```

Warning도 없어지고 deallocate도 되지 않는 상황인 것 같습니다.

2 가지 방법은 많이 다른가요? 어떤 일이 벌어진건지 잘 이해가 가지 않아 질문드립니다.
weak라서 메모리 해제된 변수의 값이 지정되면 오히려 문제인가 싶기도 하고... 시간 되실 때 답변 부탁드립니다!

Reply:
ARC 기반으로 메모리 해제되는 상황을 보려면 선언부분만 있으면 안되고, `{ }` 스코프가 같이 있어야 정확하게 볼 수 있습니다
해당 변수가 생성되서 사라지는 스코프 범위를 봐야하는거죠. 델리게이트가 weak로 지정된다는 것은 약한 참조니까 어딘가 강한 참조가 남아있어야 하는거죠. 만약 위에서 어떤게 그런 경우인지 모르겠지만, local 변수라면 스코프가 끝날 때 사라져버릴 수 있습니다.

Reply:
스코프와 어딘가 강한 참조가 남아있어야 한다는 점을 새겨야겠군요.... 확인 감사드립니다 JK!!